arm-keyvault/2015-06-01/swagger/keyvault.json

```diff
@@ -0,0 +1,463 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "KeyVaultManagementClient",
+    "description": "The Azure management API provides a RESTful set of web services that interact with Azure Key Vault.",
+    "version": "2015-06-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}": {
+      "put": {
+        "tags": [
+          "Vaults"
+        ],
+        "operationId": "Vaults_CreateOrUpdate",
+        "description": "Creates a new Azure key vault.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Resource Group to which the server belongs."
+          },
+          {
+            "name": "vaultName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the vault"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/VaultCreateOrUpdateParameters"
+            },
+            "description": "Parameters to create or update the vault"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Created or updated vault",
+            "schema": {
+              "$ref": "#/definitions/Vault"
+            }
+          },
+          "200": {
+            "description": "Created or updated vault",
+            "schema": {
+              "$ref": "#/definitions/Vault"
+            }
+          }
+        }
```

Comment:
default response is used for defining errors. Even if the default response is not specified, AutoRest generated code will deserialize the code and message properties of the error response. However, if the service is sending more information in the error response then it is recommended to define the default response for every operation. Otherwise, the extra information will not be deserialized.